internal/policy/policy_test.go

```diff
@@ -68,93 +68,195 @@ func TestInitializeNamespace(t *testing.T) {
+		if err := state.Commit(repo, "", false); err != nil {
+			t.Fatal(err)
+		}
 
-	if err := Apply(context.Background(), repo, false); err != nil {
-		t.Fatal(err)
-	}
+		if err := Apply(context.Background(), repo, false); err != nil {
+			t.Fatal(err)
+		}
 
-	targetsMetadata, err = AddDelegation(targetsMetadata, "test-rule-2", []*tuf.Key{}, []string{""}, 1)
-	if err != nil {
-		t.Fatal(err)
-	}
-	state.ruleNames.Add("test-rule-2")
+		targetsMetadata, err = AddDelegation(targetsMetadata, "test-rule-2", []*tuf.Key{}, []string{""}, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		state.ruleNames.Add("test-rule-2")
 
-	env, err = dsse.CreateEnvelope(targetsMetadata)
-	if err != nil {
-		t.Fatal(err)
-	}
+		env, err = dsse.CreateEnvelope(targetsMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	env, err = dsse.SignEnvelope(context.Background(), env, signer)
-	if err != nil {
-		t.Fatal(err)
-	}
+		env, err = dsse.SignEnvelope(context.Background(), env, signer)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	state.TargetsEnvelope = env
+		state.TargetsEnvelope = env
 
-	if err := state.Commit(repo, "", false); err != nil {
-		t.Fatal(err)
-	}
+		if err := state.Commit(repo, "", false); err != nil {
+			t.Fatal(err)
+		}
 
-	if err := Apply(context.Background(), repo, false); err != nil {
-		t.Fatal(err)
-	}
+		if err := Apply(context.Background(), repo, false); err != nil {
+			t.Fatal(err)
+		}
 
-	entry, err = rsl.GetLatestEntry(repo)
-	if err != nil {
-		t.Fatal(err)
-	}
+		entry, err = rsl.GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	loadedState, err = LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
-	if err != nil {
-		t.Error(err)
-	}
+		loadedState, err = LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
+		if err != nil {
+			t.Error(err)
+		}
 
-	assert.Equal(t, state, loadedState)
+		assert.Equal(t, state, loadedState)
+	})
+	t.Run("fail loading while verifying multiple states, bad sig", func(t *testing.T) {
+		repo, state := createTestRepository(t, createTestStateWithPolicy)
+
+		entry, err := rsl.GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		loadedState, err := LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
+		if err != nil {
+			t.Error(err)
+		}
+
+		assert.Equal(t, state, loadedState)
+
+		targetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		targetsMetadata, err = AddDelegation(targetsMetadata, "test-rule-1", []*tuf.Key{}, []string{""}, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		state.ruleNames.Add("test-rule-1")
+
+		env, err := dsse.CreateEnvelope(targetsMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		signer, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(rootKeyBytes) //nolint:staticcheck
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		badSigner, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(targets1KeyBytes) //nolint:staticcheck
```

Comment:
move this down to when it's first used, please!

Reply:
fixed